src/main/kotlin/io/github/gradlenexus/publishplugin/NexusRepository.kt

```diff
@@ -17,46 +17,36 @@
 package io.github.gradlenexus.publishplugin
 
 import org.gradle.api.Project
-import org.gradle.api.provider.Property
 import org.gradle.api.tasks.Input
-import org.gradle.api.tasks.Internal
 import org.gradle.api.tasks.Optional
 import org.gradle.kotlin.dsl.property
 import java.net.URI
 import javax.inject.Inject
 
 @Suppress("UnstableApiUsage")
-open class NexusRepository @Inject constructor(@get:Input val name: String, project: Project) {
+open class NexusRepository @Inject constructor(@Input val name: String, project: Project) {
 
-    @get:Input
-    val nexusUrl: Property<URI> = project.objects.property()
+    @Input
+    val nexusUrl = project.objects.property<URI>()
 
-    @get:Input
-    val snapshotRepositoryUrl: Property<URI> = project.objects.property()
+    @Input
```

Comment:
Tests confirms it works, but the related [documentation](https://docs.gradle.org/6.0.1/javadoc/org/gradle/api/tasks/Input.html) doesn't cover that topic:
> This annotation should be attached to the getter method in Java or the property in Groovy. Annotations on setters or just the field in Java are ignored.

Is it the recommenced approach for Kotlin?  Then documentation could be improved to: `or the property in Groovy/Kotlin`.

Reply:
I'll bring this up with the Gradle team.

Reply:
It's now an issue: https://github.com/gradle/gradle/issues/11927